### PR TITLE
Allow calling the TCI on the FD grid only every N steps, to wait M steps after rollback, and/or have K clear TCIs before going back to DG

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

I split this into a few `fixup` commits that I will squash together. Changing `SubcellOptions` is a bit tedious so it's easier to do it in one commit. However, I still wanted to keep this easy to review.

### Upgrade instructions

The TCI options now look like:

```yaml
    Subcell:
      TroubledCellIndicator:
        PerssonTci:
          Exponent: 4.0
          NumHighestModes: 1
        RdmpTci:
          Delta0: 1.0e-7
          Epsilon: 1.0e-3
        FdToDgTci:
          NumberOfStepsBetweenTciCalls: 1
          MinTciCallsAfterRollback: 1
          MinimumClearTcis: 1
        AlwaysUseSubcells: false
        UseHalo: false
        OnlyDgBlocksAndGroups: None
      SubcellToDgReconstructionMethod: DimByDim
      FiniteDifferenceDerivativeOrder: 2
```
Specifically, all the TCI-related options have been moved into a group called `TroubledCellIndicator`. The RDMP TCI options are now also in their own group. A new `FdToDgTci:` group controls when we try to go back to DG, e.g. every N steps, to wait M steps after rollback, or to have at least K clear TCIs before going back to DG. This is to allow reducing TCI overhead on FD and to allow additional smoothing by the FD solver before going back.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Evolution/DgSubcell/Actions/Test_TciAndSwitchToDg.cpp [hunk 1]
```diff
@@ -550,7 +562,7 @@ SPECTRE_TEST_CASE("Unit.Evolution.Subcell.Actions.TciAndSwitchToDg",
   register_classes_with_charm<TimeSteppers::AdamsBashforth,
                               TimeSteppers::Rk3HesthavenSsp>();
   test<1>();
-  test<2>();
-  test<3>();
+  // test<2>();
```
Comment (kidder) [inline tests/Unit/Evolution/DgSubcell/Actions/Test_TciAndSwitchToDg.cpp]: why are these commented out?

File: tests/Unit/Evolution/DgSubcell/Actions/Test_TciAndSwitchToDg.cpp [hunk 2]
```diff
@@ -605,7 +616,7 @@ SPECTRE_TEST_CASE("Unit.Evolution.Subcell.Actions.TciAndSwitchToDg",
   register_classes_with_charm<TimeSteppers::AdamsBashforth,
                               TimeSteppers::Rk3HesthavenSsp>();
   test<1>();
-  // test<2>();
-  // test<3>();
+  test<2>();
```
Comment (kidder) [inline tests/Unit/Evolution/DgSubcell/Actions/Test_TciAndSwitchToDg.cpp]: ah never mind

**Review by kidder — COMMENTED** [review]
LGTM

**Comment by nilsdeppe** [comment]
Rebased and squashed. Thanks for the review!

**Review by kidder — APPROVED** [review]

**Pull request merged**